src/app/services/storage.service.ts

```diff
@@ -240,4 +240,16 @@ export class StorageService {
 
     return path.indexOf('/') < 0;
   }
+
+  // Takes a string with fuzzy units like 20Mib, 20k, 20Gb, returns number in bytes
+  convertDataMeasurementStrings(str: string) {
+    if(!str) { return 0 }; // if empty, return the default 0, which is unlimited
+    if(str[0] && !str[0].match(/[0-9]/)) { return NaN };
+    const letters = str.replace( /[^a-zA-Z]/g, '');
+    if (letters[0] && !letters[0].toLowerCase().match(/b|k|m|g|t/i)) { return NaN };
+    const powersOf1024 = {b: 1, k: 1024, m: 1024**2, g: 1024**3, t: 1024**4};
+    const unit = str.toLowerCase().match(/[b|k|m|g|t]/i) || 'b';
```

Comment:
Neat!  That's a nice way to set a default.  Might as well remove the "i" flag, since it's not being used.
```suggestion
    const unit = str.toLowerCase().match(/[b|k|m|g|t]/) || 'b';
```